dbt_subprojects/solana/models/_sector/staking/solana/staking_solana_stake_account_delegations.sql

```diff
@@ -0,0 +1,78 @@
+    , alias = 'stake_account_delegations'
+    , materialized = 'incremental'
+    , file_format = 'delta'
+    , incremental_strategy = 'merge'
+    , unique_key = ['block_time', 'block_slot', 'stake_account', 'vote_account']
+    , incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.block_time')]
+    , post_hook='{{ expose_spells(\'["solana"]\',
+                                "sector",
+                                "staking",
+                                \'["ilemi"]\') }}')
+}}
+
+with 
+    all_delegates as (
+        --get all stake accounts, and get all of the vote accounts they have been delegated to over time
+        SELECT
+            distinct
+            account_stakeAccount
+            , account_voteAccount
+            , call_block_time
+            , call_block_slot
+        FROM (
+            SELECT 
+                account_stakeAccount
+                , account_voteAccount
+                , COALESCE(call_block_time, timestamp '1990-01-01 00:00:00') as call_block_time --fill in nulls, at a fake earlier date than split thing below
+                , call_block_slot
+            FROM {{ source ('stake_program_solana','stake_call_DelegateStake') }}
+            {% if is_incremental() %}
+            WHERE {{incremental_predicate('call_block_time')}}
+            {% endif %}
+
+            UNION ALL
+            
+            --SPLIT creates a new account, but copies vote delegation from original. We get that and also look for the latest delegation of the account.
+            SELECT 
+                a.destination as account_stakeAccount
+                , COALESCE(del.account_voteAccount, old_vote.account_voteAccount) as account_voteAccount
+                , COALESCE(del.call_block_time, old_vote.call_block_time) as call_block_time
+                , COALESCE(del.call_block_slot, old_vote.call_block_slot) as call_block_slot
+            FROM {{ ref('staking_solana_stake_actions') }} a
+            --get inherited vote account
+            LEFT JOIN (SELECT 
+                            source
+                            , del.account_voteAccount
+                            , del.filled_block_time as call_block_time --fill in nulls with fake date
+                            , del.filled_block_slot as call_block_slot
+                        FROM {{ ref('staking_solana_stake_actions') }} a
+                        LEFT JOIN (SELECT *
+                                        , COALESCE(call_block_time, timestamp '2000-01-01 00:00:00') as filled_block_time
+                                        , call_block_slot as filled_block_slot
+                                    FROM {{ source ('stake_program_solana','stake_call_DelegateStake') }}
+                                    ) del
+                            ON del.account_stakeAccount = a.source
+                            AND del.filled_block_time <= a.block_time
+                        WHERE action = 'split'
+                    ) as old_vote
```

Comment:
since the main query is filtering incrementally, should this subquery that is left joined also include incremental filters? or does it need history?

Reply:
this needs history unfortunately. we could move it into a sub incremental model if we wanted to